easybuild/easyblocks/g/gcc.py

```diff
@@ -354,10 +394,15 @@ def configure_step(self):
         if self.cfg['languages']:
             self.configopts += " --enable-languages=%s" % ','.join(self.cfg['languages'])
 
-        if self.cfg['withnvptx']:
-            if self.iter_idx == 0:
+        if self.cfg['withnvptx'] or self.cfg['withamdgcn']:
+            if self.build_stages[self.iter_idx] == HOST_COMPILER:
                 self.configopts += " --without-cuda-driver"
-                self.configopts += " --enable-offload-targets=nvptx-none"
+                offload_target = []
+                if self.cfg['withnvptx']:
+                    offload_target += ['nvptx-none']
+                if self.cfg['withamdgcn']:
+                    offload_target += ['amdgcn-amdhsa']
+                self.configopts += " --enable-offload-targets=%s" % ','.join(offload_target)
```

Comment:
Does building both work? What if there are none, should raise an error? (just to protect against modifying the `if` above, but do something silly here)

Reply:
Building both works as expected, have not tested the Nvidia offloading once both are built so that is something to check

Reply:
Just tested the build on an Nvidia machine and it works there as well